packages/server/graphql/mutations/helpers/generateWholeMeetingSummary.ts

```diff
@@ -0,0 +1,36 @@
+import getRethink from '../../../database/rethinkDriver'
+import OpenAIServerManager from '../../../utils/OpenAIServerManager'
+import {DataLoaderWorker} from '../../graphql'
+import isValid from '../../isValid'
+
+const generateWholeMeetingSummary = async (
+  discussionIds: string[],
+  meetingId: string,
+  facilitatorUserId: string,
+  dataLoader: DataLoaderWorker
+) => {
+  const [facilitator, commentsByDiscussions, tasksByDiscussions, reflections] = await Promise.all([
+    dataLoader.get('users').loadNonNull(facilitatorUserId),
+    dataLoader.get('commentsByDiscussionId').loadMany(discussionIds),
+    dataLoader.get('tasksByDiscussionId').loadMany(discussionIds),
+    dataLoader.get('retroReflectionsByMeetingId').load(meetingId)
+  ])
+  if (!facilitator.featureFlags.includes('aiSummary')) return
+  const manager = new OpenAIServerManager()
+  const reflectionsContent = reflections.map((reflection) => reflection.plaintextContent)
+  const commentsContent = commentsByDiscussions
+    .filter(isValid)
+    .flatMap((commentsByDiscussion) =>
+      commentsByDiscussion.map(({plaintextContent}) => plaintextContent)
+    )
+  const tasksContent = tasksByDiscussions
+    .filter(isValid)
+    .flatMap((tasksByDiscussion) => tasksByDiscussion.map(({plaintextContent}) => plaintextContent))
+  const contentToSummarize = [...commentsContent, ...tasksContent, ...reflectionsContent]
```

Comment:
Just wonder if AI can summarize better, if the content is somehow divided into groups, so he know that the comments belong to the specific topic

Reply:
We already have the discussion summaries (which are made up of all the comments and tasks), and we have the topic summaries (which are the summaries of the reflections), so instead of summarising all comments, tasks, and reflections, we could just summarise all of the topic and discussion summaries? 

We could see how this works in production with real meeting content and then make the change if the summaries don't seem very good?

Reply:
Summarizing summaries is an interesting idea.

> We could see how this works in production with real meeting content and then make the change if the summaries don't seem very good?

Make sense!